src/state/slices/index.ts

```diff
@@ -7,3 +66,72 @@ export * from "./scatterplot_slice";
 export * from "./threshold_slice";
 export * from "./time_slice";
 export * from "./vector_slice";
+
+export type ViewerStoreState = BackdropSliceState &
+  CollectionSliceState &
+  ColorRampSliceState &
+  ConfigSliceState &
+  DatasetSliceState &
+  ScatterPlotSliceState &
+  ThresholdSliceState &
+  TimeSliceState &
+  VectorSliceState;
+
+export type ViewerStoreActions = BackdropSliceActions &
+  CollectionSliceActions &
+  ColorRampSliceActions &
+  ConfigSliceActions &
+  DatasetSliceActions &
+  ScatterPlotSliceActions &
+  ThresholdSliceActions &
+  TimeSliceActions &
+  VectorSliceActions;
```

Comment:
Explicitly separated state and actions, which are combined to form the final store.